src/poprox_recommender/components/filters/image_selector.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import Optional
+
+import torch as th
+
+from poprox_concepts.domain import Image, InterestProfile, RecommendationList
+from poprox_recommender.components.embedders.caption_embedder import CaptionEmbedder
+from poprox_recommender.pytorch.decorators import torch_inference
+
+logger = logging.getLogger(__name__)
```

Comment:
I'd suggest either uncommenting the debug logging below and set the log level on this logger to info or warning to hide them, or removing the commented log statements.